### PR TITLE
[WIP] [jaeger-v2] Refactor Configurations For Adaptive Sampling Processor

### LINKED ISSUE #6021: [jaeger-v2] Refactor Adaptive Sampling Processor Configurations
### Requirement

In this issue, we want to refactor the configurations for the adaptive sampling processor to ensure that the configuration is laid out logically (part of https://github.com/jaegertracing/jaeger/issues/5229)

### Tasks / outcomes

- [x] All config fields are tagged with mapstructure such that they can be addressable from YAML via "good" names
- [x] Configs implement Validate()
- [x] Configs via YAML support the same defaults as in jaeger-v1
- [x] Configs reuse standard elements of OTEL configs whenever possible, e.g. TLS, clientcfg, etc.
- [x] Configuration looks "logical" and properly grouped, not flattened with long weird names
- [x] We create a migration guide document with a table that lists v1 CLI flags and the corresponding v2 config fields

### PR DESCRIPTION
**Pull request by mahadzaryab1**
## Which problem is this PR solving?
- Resolves #6021

## Description of the changes
- [Migration Guide](https://docs.google.com/document/d/1_gtjUP3FlFTY5Ni98fP9qN9TnGCCQnMTQuyew_xEQAc/edit)

## How was this change tested?
- CI / Unit Tests / Integration Tests

## Checklist
- [x] I have read https://github.com/jaegertracing/jaeger/blob/master/CONTRIBUTING_GUIDELINES.md
- [x] I have signed all commits
- [x] I have added unit tests for the new functionality
- [x] I have run lint and test steps successfully
  - for `jaeger`: `make lint test`
  - for `jaeger-ui`: `yarn lint` and `yarn test`

### CODE REVIEW

File: plugin/sampling/strategyprovider/adaptive/options.go [hunk 1]
```diff
@@ -40,34 +40,30 @@ const (
...
+	// to reduce lock thrashing.
+	LeaderLeaseRefreshInterval time.Duration `mapstructure:"leader_lease_refresh_interval"`
 
-	// CalculationInterval determines how often new probabilities are calculated. E.g. if it is 1 minute,
-	// new sampling probabilities are calculated once a minute and each bucket will contain 1 minute worth
-	// of aggregated throughput data.
-	CalculationInterval time.Duration `mapstructure:"calculation_interval"`
+	// FollowerLeaseRefreshInterval is the duration to sleep if this processor is a follower
+	// (ie. failed to gain the leader lock).
+	FollowerLeaseRefreshInterval time.Duration `mapstructure:"follower_lease_refresh_interval"`
```
Comment (mahadzaryab1) [inline plugin/sampling/strategyprovider/adaptive/options.go:53]: @yurishkuro do you have any thoughts on how to group these? I was thinking grouping them into a type like `Synchronization`.

File: cmd/jaeger/config.yaml [hunk 2]
```diff
@@ -38,7 +38,8 @@ extensions:
     #   path: ./cmd/jaeger/sampling-strategies.json
     adaptive:
       sampling_store: some_store
-      initial_sampling_probability: 0.1
+      sampling:
```
Comment (yurishkuro) [inline cmd/jaeger/config.yaml:41]: The whole struct here is "adaptive sampling", I think repeating "sampling" is redundant, does not add any clarity.

File: plugin/sampling/strategyprovider/adaptive/options.go [hunk 3]
```diff
@@ -40,34 +40,30 @@ const (
 // The abbreviation SPS refers to "samples-per-second", which is the target
 // of the optimization/control implemented by the adaptive sampling.
 type Options struct {
-	// TargetSamplesPerSecond is the global target rate of samples per operation.
-	// TODO implement manual overrides per service/operation.
-	TargetSamplesPerSecond float64 `mapstructure:"target_samples_per_second"`
+	Calculation Calculation `mapstructure:"calculation"`
+	Sampling    Sampling    `mapstructure:"sampling"`
```
Comment (yurishkuro) [inline plugin/sampling/strategyprovider/adaptive/options.go:44]: I'm not sure this is an improvement. Most flags are related to calculation anyway. Some are related to boundary conditions, like min rate.

**Review by yurishkuro — COMMENTED** [review]
I'm not seeing this change as an improvement. The new subgroupings are not logical or better than a flat list. Maybe this config is already good as is.

**Comment by mahadzaryab1** [comment]
> I'm not seeing this change as an improvement. The new subgroupings are not logical or better than a flat list. Maybe this config is already good as is.

@yurishkuro Here is what the current full config would look like
```yaml
  remote_sampling:
    adaptive:
      sampling_store: some_store
      target_samples_per_second: 
      delta_tolerance: 
      calculation_interval: 
      aggregation_buckets: 
      calculation_buckets: 
      calculation_delay: 
      initial_sampling_probability: 
      min_sampling_probability: 
      min_samples_per_second: 
      leader_lease_refresh_interval: 
      follower_lease_refresh_interval: 
    http:
    grpc:
```
I saw two main groupings here. One for the collection of the data (sampling), and one for the computations (calculation). The config from the current PR would look something like:
```yaml
  remote_sampling:
    adaptive:
      calulation: 
        aggregation_buckets:
        buckets:
        delay:
        delta_tolerance:
        interval:
      sampling:
        store: 
        initial_probability:
        min_probability: 
        min_rate:
        target_rate: 
      # potentially a better grouping for the following two
      leader_lease_refresh_interval: 
      follower_lease_refresh_interval: 
    http:
    grpc:
```
Let me know what you think. Do you see any other groupings or do you want to keep the current flat list? I can adjust the migration guide accordingly and close this PR out if we don't want any changes here.

**Comment by yurishkuro** [comment]
I find the flat list easier to read right now. The calc/sampling separation is artificial, looks like it saves typing one word, but actually makes naming harder to understand. I don't think it's worth changing.

**Comment by mahadzaryab1** [comment]
Closing as per discussion above

**Pull request closed without merging**